### PR TITLE
Asset representative

### PR DESCRIPTION
**Pull request by cuteolaf**
- Added `representative` field in the asset struct
- Added more checks for `launch_representative_session`
- Updated the functions and tests accordingly

### CODE REVIEW

File: pallets/asset_management/src/tests.rs [hunk 1]
```diff
@@ -479,7 +487,7 @@ fn test_integration_test() {
 				VoteProposals::Election,
 				Ident::Judgement::Reasonable,
 			),
-			Error::<Test>::AlreadyLinkedWithAsset
+			Error::<Test>::RepresentativeAlreadyLinkedWithAsset
```
Comment (ndkazu) [inline pallets/asset_management/src/tests.rs:490]: This is a tenant session, not a Representative_session

File: pallets/asset_management/src/tests.rs [hunk 2]
```diff
@@ -75,9 +75,13 @@ pub fn prep_test(
 	));
 }
 
-// FIXME: write more tests for the representative flow
 #[test]
 fn test_representative() {
+	// TODO: write more tests for the representative flow
+	// edge cases to cover
```
Comment (ndkazu) [inline pallets/asset_management/src/tests.rs:81]: Please, see the test in the Tenancy_pallet: Representative already linked to an asset get assigned a second one in a second referendum.

**Review by ndkazu — CHANGES_REQUESTED** [review]
Tenant_session error is not right.

File: pallets/asset_management/src/lib.rs [hunk 3]
```diff
@@ -251,12 +251,18 @@ pub mod pallet {
...
+		AssetNotLinkedWithRepresentative,
+		/// The given representative is not linked with the asset
+		InvalidRepresentative,
 		/// The asset is not linked to the representative
 		AssetOutOfControl,
 		/// The candidate is not a tenant
 		NotATenant,
 		/// An asset is already linked to the provided account
-		AlreadyLinkedWithAsset,
+		RepresentativeAlreadyLinkedWithAsset,
```
[the remark this answers is not in the document]
Comment (cuteolaf) [reply pallets/asset_management/src/lib.rs:265]: nice catch, i fixed it in the other PR.

File: pallets/asset_management/src/tests.rs (hunk 1 above)
Comment (cuteolaf) [reply pallets/asset_management/src/tests.rs:490]: nice catch, i fixed it in the other PR.
